Author: Marcin Juszkiewicz <marcin.juszkiewicz@linaro.org>
Description: fix cross build

When klibc is cross built we want /usr/include/DEB_HOST_MULTIARCH/asm as
include/asm not /usr/include/DEB_BUILD_MULTIARCH/asm one.

---
 scripts/Kbuild.install |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- klibc-1.5.25.orig/scripts/Kbuild.install
+++ klibc-1.5.25/scripts/Kbuild.install
@@ -93,16 +93,16 @@ header:
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)
 	$(Q)-rm -rf $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)lib
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)bin
-	$(Q)if [ -n "$(DEB_HOST_MULTIARCH)" ]; then \
-		ln -s /usr/include/$(DEB_HOST_MULTIARCH)/asm $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/ || exit; \
-	fi
 	$(Q)for x in /usr/include/linux /usr/include/asm*; do \
 		ln -s $${x} $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/ || exit; \
 	done
+	$(Q)if [ -n "$(DEB_HOST_MULTIARCH)" ]; then \
+		ln -sf /usr/include/$(DEB_HOST_MULTIARCH)/asm $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/ || exit; \
+	fi
 	$(Q)cp -rf usr/include/. $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/.
 	$(Q)chmod -R a+rX $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)$(install-data) $(srctree)/klcc/klcc.1 $(INSTALLROOT)$(mandir)/man1/$(KCROSS)klcc.1
 	$(Q)$(install-bin) $(objtree)/klcc/$(KCROSS)klcc $(INSTALLROOT)$(bindir)
 
